internal/querynode/segment_loader.go

```diff
@@ -517,48 +517,72 @@ func (loader *segmentLoader) FromDmlCPLoadDelete(ctx context.Context, collection
 	defer stream.Close()
 	pChannelName := rootcoord.ToPhysicalChannel(position.ChannelName)
 	position.ChannelName = pChannelName
-	stream.AsReader([]string{pChannelName}, fmt.Sprintf("querynode-%d-%d", Params.QueryNodeCfg.QueryNodeID, collectionID))
+	hasMore := stream.AsConsumerToLastMsg(pChannelName, fmt.Sprintf("querynode-%d-%d", Params.QueryNodeCfg.QueryNodeID, collectionID))
```

Comment:
My proposal is to simply the interface with only one interface -> GetLatestMsg

	latestPosition = stream.GetLatestMsg 
	do AsConsumer
        do seek
	for _, tsMsg := range msgPack.Msgs {
		if tsMsg.Type() == commonpb.MsgType_Delete {
			dmsg := tsMsg.(*msgstream.DeleteMsg)
			if dmsg.CollectionID != collectionID {
				continue
			}
			log.Debug("delete pk",
				zap.Any("pk", dmsg.PrimaryKeys),
				zap.String("vChannelName", position.GetChannelName()),
				zap.Any("msg id", position.GetMsgID()),
			)
			processDeleteMessages(loader.historicalReplica, dmsg, delData)
		}

		if (tsMsg.Position() >= latestPosition) {
			hasMore = true
			break
		}
	}